macOS/DuckDuckGo/Feedback/Model/Feedback.swift

```diff
@@ -35,5 +35,80 @@ struct Feedback {
     let comment: String
     let appVersion: String
     let osVersion: String
+    let subcategory: String
 
+    init(category: Category,
+         comment: String,
+         appVersion: String,
+         osVersion: String,
+         subcategory: String = "") {
+        self.category = category
+        self.comment = comment
+        self.appVersion = appVersion
+        self.osVersion = osVersion
+        self.subcategory = subcategory
+    }
+}
+
+extension Feedback {
+
+    static func from(selectedPills: [String],
+                     text: String,
+                     appVersion: String,
+                     category: Feedback.Category,
+                     problemCategory: ProblemCategory?) -> Feedback {
+        let selectedOptionsString = selectedPills.map { $0.toTag }.joined(separator: ",")
+        let description = text.isEmpty ? category.toString : text
+        var subcategory = ""
+
+        if let problemCategory = problemCategory {
+            subcategory = "\(problemCategory.name.toTag),\(selectedOptionsString)"
+        } else {
+            subcategory = "\(selectedOptionsString)"
+        }
+
+        return Feedback(category: category,
+                        comment: description,
+                        appVersion: appVersion,
+                        osVersion: "\(ProcessInfo.processInfo.operatingSystemVersion)",
+                        subcategory: subcategory)
+    }
+}
+
+extension String {
+    var toTag: String {
```

Comment:
The purpose of this function is to sanitize the options in the Request New Feature and Report a Browser Problem to an Asana tag. This will convert the text into a more legible form to use as an Asana tag.

Examples:
“Browser install & updates” -> “browser-install-and-updates”
“Camera/audio permissions” -> “camera-audio-permissions”
“Video doesn't play” -> “video-doesnt-play"